src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -159,6 +160,7 @@ OverlayImpl::OverlayImpl (
     , m_resolver (resolver)
     , next_id_(1)
     , timer_count_(0)
+    , m_stats(std::bind (&OverlayImpl::collect_metrics, this),collector)
```

Comment:
There's an alternative here which allows you to use (the heretofore unused) `Application::getCollectorManager` function:

```
    , m_stats(
        std::bind(&OverlayImpl::collect_metrics, this),
        app_.getCollectorManager().collector())
```

This _slightly_ simplifies `OverlayImpl` construction, but I'm not sure if it's a net gain. All these are codepaths we call once and only once. It also doesn't help that we never use `Application::getCollectorManager` anywhere currently, which means we could (_should_?) remove it, slimming down `Application`.

Reply:
I think we should remove getCollectorManager since it's unused. If we decided to start using it, we would need to make changes wherever we pass the collector in through the constructor. I'm happy to do that, but I don't think it would really be a net gain.